tools/bin/commands/util/openshift_funcs

```diff
@@ -222,3 +222,42 @@ guess_route() {
     fi
     echo "${project}.${domain#*.}"
 }
+
+create_api_cookie() {
+
+    local COOKIE1="$HOME/syndesis_cookie.txt"
+    # second cookie needed because curl flushes old values upon redirect. Might be a bug or some obscure flag missing
+    local COOKIE2="$(mktemp /tmp/syndesis_cookie.XXX.txt)"
+    # generate a new user so that it will be asked to approve permissions delegation
+    local USERNAME="dev$RANDOM"
+
+    # clean cookie
+    rm -rf "$COOKIE1"
+    touch "$COOKIE1"
+
+    local SYNDESIS_URL="$(oc get route syndesis  --template={{.spec.host}})"
+    # first invocation is needed just to get the redierect url and to define a csrf token
+    local REDIRECTED_URL="$(curl --write-out "%{url_effective}\n" --location --silent --show-error --output /dev/null --insecure --cookie-jar  "$COOKIE1" -v "https://$SYNDESIS_URL" 2>/dev/null )"
+
+    # isolate required params
+    local LOGIN_URL="$(echo $REDIRECTED_URL | sed -e 's/?/\n/g' | head -n1)"
+    local THEN_URL="$(echo $REDIRECTED_URL | sed -e 's/=/\n/g' | tail -n1)"
```

Comment:
are you sure that the parameter value you want to pick up is always last ? Looks quite fragile to me, maybe better to use a regexp on the parameter key with a capturing group around the value ?